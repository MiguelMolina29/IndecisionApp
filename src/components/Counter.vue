<template>
    <h2>{{customTitle}}</h2>
    <p> {{counter}} <sup>2</sup> = {{squareCounter}} </p>

    <div>
        <button @click="incrementar()">+1</button>
        <button @click="decrementar()">-1</button>
    </div>

</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 10, // Esto da un numero por defecto
            //required: true // esto le dice a vue que es obligatorio
            validator( value ) {
                return value >= 10
            }
        }
    },
    data() { // La data regresa un objeto 
        return {
            counter: this.start,
        }
    },
    methods: {
        getSquareValue() {
            return this.counter * this.counter
        },
        incrementar() {
            return this.counter = this.counter + 1
        },
        decrementar() {
            return this.counter = this.counter - 1
        }
    },
    computed: {
        squareCounter() {
            return this.counter * this.counter
        },
        customTitle() {
            return !this.title ? 'counter' : this.title
        }
    }
}
</script>

<style scoped>
    button{ 
        background-color: #64BB87;
        border-radius: 5px;
        border: 1px solid white;
        color: white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover {
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }
</style>