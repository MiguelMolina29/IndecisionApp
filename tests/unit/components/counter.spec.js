import {shallowMount} from '@vue/test-utils'
import Counter from '@/components/Counter';

describe('Counter Component', () => {
    
    // test('Debe hacer match con el snapshot', () => {
        
    //     const wrapper = shallowMount(Counter) // wrapper es un standar

    //     expect(wrapper.html()).toMatchSnapshot()

    // })

    // Segunda prueba
    test('H2 debe tener el valor por defecto de "Counter"', () => {
        
        const wrapper = shallowMount( Counter );

        const h2 = wrapper.find('h2')

        expect(h2.text()).toBe('counter')

    })
    
    
})
