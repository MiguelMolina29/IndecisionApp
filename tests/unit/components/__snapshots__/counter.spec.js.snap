// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe hacer match con el snapshot 1`] = `
<h2>counter</h2>
<p>10 <sup>2</sup> = 100</p>
<div><button>+1</button><button>-1</button></div>
`;
